mypy/semanal.py

```diff
@@ -5354,7 +5376,13 @@ def analyze_try_stmt(self, s: TryStmt, visitor: NodeVisitor[None]) -> None:
                 type.accept(visitor)
             if var:
                 self.analyze_lvalue(var)
-            handler.accept(visitor)
+            if s.is_star:
+                was_in_except_block: bool = self.is_in_except_block
+                self.is_in_except_block = True
+                handler.accept(visitor)
+                self.is_in_except_block = was_in_except_block
```

Comment:
To save on some repeated boilerplate, we could manage this flag with a context manager, similar to the existing [`allow_unbound_tvars_set`](https://github.com/python/mypy/blob/3b00002acdf098e7241df8f2e1843f8b8260b168/mypy/semanal.py#L506).